server/src/test/java/org/opensearch/search/aggregations/startree/BoolStarTreeFilterProviderTests.java

```diff
@@ -0,0 +1,894 @@
+            .must(new BoolQueryBuilder().must(new BoolQueryBuilder()));
+
+        StarTreeFilterProvider provider = StarTreeFilterProvider.SingletonFactory.getProvider(boolQuery);
+        StarTreeFilter filter = provider.getFilter(searchContext, boolQuery, compositeFieldType);
+
+        assertNull("Filter should be null for empty nested bool queries", filter);
+    }
+
+    public void testSingleClauseBoolQueries() throws IOException {
+        // Test single MUST clause
+        BoolQueryBuilder mustOnly = new BoolQueryBuilder().must(new TermQueryBuilder("status", 200));
+
+        StarTreeFilterProvider provider = StarTreeFilterProvider.SingletonFactory.getProvider(mustOnly);
+        StarTreeFilter filter = provider.getFilter(searchContext, mustOnly, compositeFieldType);
+
+        assertNotNull("Filter should not be null", filter);
+        assertEquals("Should have one dimension", 1, filter.getDimensions().size());
+        assertExactMatchValue((ExactMatchDimFilter) filter.getFiltersForDimension("status").get(0), 200L);
+
+        // Test single SHOULD clause
+        BoolQueryBuilder shouldOnly = new BoolQueryBuilder().should(new TermQueryBuilder("status", 200));
+
+        filter = provider.getFilter(searchContext, shouldOnly, compositeFieldType);
+
+        assertNotNull("Filter should not be null", filter);
+        assertEquals("Should have one dimension", 1, filter.getDimensions().size());
+        assertExactMatchValue((ExactMatchDimFilter) filter.getFiltersForDimension("status").get(0), 200L);
+    }
+
+    public void testDuplicateDimensionsAcrossNesting() throws IOException {
+        // Test duplicate dimensions that should be merged/intersected
+        BoolQueryBuilder boolQuery = new BoolQueryBuilder().must(new RangeQueryBuilder("status").gte(200).lte(500))
+            .must(new BoolQueryBuilder().must(new RangeQueryBuilder("status").gte(300).lte(400)));
+
+        StarTreeFilterProvider provider = StarTreeFilterProvider.SingletonFactory.getProvider(boolQuery);
+        StarTreeFilter filter = provider.getFilter(searchContext, boolQuery, compositeFieldType);
+
+        assertNotNull("Filter should not be null", filter);
+        assertEquals("Should have one dimension", 1, filter.getDimensions().size());
+
+        List<DimensionFilter> statusFilters = filter.getFiltersForDimension("status");
+        assertEquals("Should have one filter after intersection", 1, statusFilters.size());
+        RangeMatchDimFilter rangeFilter = (RangeMatchDimFilter) statusFilters.getFirst();
+        assertEquals("Lower bound should be 300", 300L, rangeFilter.getLow());
+        assertEquals("Upper bound should be 400", 400L, rangeFilter.getHigh());
+        assertTrue("Lower bound should be exclusive", rangeFilter.isIncludeLow());
+        assertTrue("Upper bound should be exclusive", rangeFilter.isIncludeHigh());
+    }
+
+    public void testKeywordFieldTypeHandling() throws IOException {
```

Comment:
Can we please add one test case for keyword range? Also for floats.

Reply:
Added including float intersection tests